code/src/sixsq/nuvla/server/resources/infrastructure_service/utils.clj

```diff
@@ -0,0 +1,21 @@
+(ns sixsq.nuvla.server.resources.infrastructure-service.utils
+  (:require
+    [sixsq.nuvla.server.resources.common.schema :as c]
+    [sixsq.nuvla.server.util.response :as r]))
+
+
+(defn can-delete?
+  [{:keys [state] :as resource}]
+  (#{"CREATED" "STOPPED"} state))
+
+
+(defn verify-can-delete
+  [{:keys [id state] :as resource}]
+  (if (can-delete? resource)
+    resource
+    (throw (r/ex-response (str "invalid state (" state ") for delete on " id) 412 id))))
+
+
+(defn remove-delete
```

Comment:
`remove-delete-op` would be a better name for this function.

Reply:
don't need this function